imageregistry/v1/types.go

```diff
@@ -306,6 +306,81 @@ type ImageRegistryConfigStorageIBMCOS struct {
 	ServiceInstanceCRN string `json:"serviceInstanceCRN,omitempty"`
 }
 
+// EndpointAccessibility defines the Alibaba VPC endpoint for storage
+type EndpointAccessibility string
+
+// AlibabaEncryptionMode defines an enumerable type for the encryption mode
+type AlibabaEncryptionMode string
+
+const (
+	// InternalEndpoint sets the VPC endpoint to internal
+	InternalEndpoint EndpointAccessibility = "Internal"
+	// PublicEndpoint sets the VPC endpoint to public
+	PublicEndpoint EndpointAccessibility = "Public"
+
+	// PlainText is an AlibabaEncryptionMode. This is the default. This means no encryption
+	PlainText AlibabaEncryptionMode = "PlainText"
+	// AES256 is an AlibabaEncryptionMode. This means AES256 encryption
+	AES256 AlibabaEncryptionMode = "AES256"
+	// KMS is an AlibabaEncryptionMode. This means KMS encryption
+	KMS AlibabaEncryptionMode = "KMS"
+)
+
+// EncryptionAlibaba this a union type in kube parlance.  Depending on the value for the encryptionType,
+// different pointers may be used
+type EncryptionAlibaba struct {
+	// Mode defines the different encrytion types available
+	// Empty value means no opinion and the platform chooses the a default, which is subject to change over time.
+	// Currently the default is `AES256`.
+	// +kubebuilder:validation:Enum="PlainText";"KMS";"AES256"
+	// +kubebuilder:default="AES256"
+	// +optional
+	Mode AlibabaEncryptionMode `json:"mode"`
```

Comment:
@sttts @kwoodson I am concerning about the field name `type`, looking the Alibaba Console, bucket encryption configuration, we have three **methods** of encryption:
- None
- OSS-Managed
- KMS

Then it has only one encryption **algorithm**:
- AES256

I am concerned if we could rename this field from `type` to `method`, with the default value of `OSS-Managed`.

I am not sure if we need the field `algorithm`, but it could be interesting for future capabilities. For now it will assume only one value: `AES256`

![Screenshot from 2021-12-14 15-34-32](https://user-images.githubusercontent.com/3216894/146062003-647237e1-bc0a-4394-959b-c61f8ce725d2.png)
![Screenshot from 2021-12-14 15-34-26](https://user-images.githubusercontent.com/3216894/146062007-a529642b-82fc-47b8-8d09-86e211bfaa35.png)

Reply:
I updated the `Type` to `Mode` to align with the API. I updated the default to `AES256` after @mtulio's tests. This seems to be what we wanted it to be by default.

Reply:
Here is the reference where we are setting to default, if KMS is not set:
https://github.com/openshift/cluster-image-registry-operator/pull/724/files#diff-e0f51602f2f4156d47e688910139da09ff81fe2ac10efa43d7d453b06ee08ec8R502

Reply:
Have nothing against these changes, but be quick. This should land before FF on Friday (I am out, but David Eads can approve).

Reply:
@sttts Looks like we have been bitten by the `yaml_patch` issue. I have submitted a PR #1086 to fix it. We would like to merge that one so that I can generate code properly in this PR. Would you please TAL?

Reply:
cc @deads2k

Reply:
tagged

Reply:
@sttts Awesome, thanks for your help!